### ISSUE TITLE
RangeRate seems to be incorrect

### ISSUE THREAD

**Issue by Magner18** [parzivail/SGP.NET#25]
I Implemented a very basic application using SGP.NET and I think I am getting incorrect range rate values. my little test app is tracking STARLINK-3962, and I can watch the range change by something like .2 km/s or 1km/s depending on the satellites current position relative to the ground station. anyway the reported range rate is sometime 0km/s or 0.00000000000000002079601643302110 km/s. anyway that value I think is incorrect.

**Comment by parzivail** [comment]
Can you share the TLE and a timestamp where the issue occurs? I'll get it set up in a debugger and see where the issue is coming from. I'll also compare it to gpredict since the RR calculation is using the same formula.

**Comment by Magner18** [comment]
that also breaks. I think the issue may be that by setting time to DateTime.Now or DateTime.UtcNow the value changes while it is doing the calculations.

**Comment by Magner18** [comment]
Nope, thats not it, I think it just something to do with a time generated using DateTime.Now

**Comment by Magner18** [comment]
ok I created the time var like this
var time = new DateTime(Datetime.Now.Year, Datetime.Now.Month, Datetime.Now.Day, Datetime.Now.Hour, Datetime.Now.Minute, Datetime.Now.Second, DateTimeKind.UTC);

and the problem went away.

**Comment by Magner18** [comment]
4.0.30319.42000

**Comment by Magner18** [comment]
@parzivail so you are able to reproduce the issue I am seeing?

**Comment by parzivail** [comment]
Apologies for the delay -- No, I haven't been able to reproduce this issue, on .NET 4 or .NET 6. Can you put together a minimal repo that can demonstrate the issue so I can reproduce your setup exactly? I doubt this is a runtime issue but it's very strange that using the dynamic `DateTime` fields break but using a constructed `DateTime` works.
